pixels-retina/src/main/java/io/pixelsdb/pixels/retina/PixelsWriterBuffer.java

```diff
@@ -369,27 +372,28 @@ private void startFlushMinioToDiskScheduler(long intervalSeconds)
                 }
             } catch (Exception e)
             {
-                throw new RuntimeException(e);
+                logger.error("Failed to flush data to disk", e);
             }
         }, 0, intervalSeconds, TimeUnit.SECONDS);
     }
 
     /**
-     * collect resources
-     *
-     * @throws RetinaException
+     * Gracefully close the writer buffer, ensuring all in-memory data is persisted.
      */
     public void close() throws RetinaException
     {
         // First, shut down the flush process to prevent changes to the data view.
         this.flushMinioExecutor.shutdown();
         try
         {
-            this.flushMinioExecutor.awaitTermination(Long.MAX_VALUE, TimeUnit.NANOSECONDS);
+            if (!this.flushMinioExecutor.awaitTermination(60, TimeUnit.SECONDS))
+            {
+                this.flushMinioExecutor.shutdown();
```

Comment:
Shutdown only stops accepting new threads; it does not terminate the threads.
Please check if shutdownNow should be used here instead.